File: C++/Easy/Accepted/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence_6.cpp
class Solution {
public:
    int isPrefixOfWord(string sentence, string searchWord) {
        
        int index = 1;
        string word;
        string::size_type pos;
        while ((pos = sentence.find(' ')) != string::npos) {
            word = sentence.substr(0, pos);
            if (word.find(searchWord) == 0) {
                return index;
            }
            sentence.erase(0, pos + 1);
            index++;
        }
        if (sentence.find(searchWord) == 0) {
            return index;
        }
        return -1;
    }
};

File: Rust/Easy/Accepted/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence_6.rs
impl Solution {
    pub fn is_prefix_of_word(sentence: String, search_word: String) -> i32 {
        
        let mut index = 0;
        for word in sentence.split(' ') {
            index += 1;
            if word.starts_with(&search_word) {
                return index;
            }
        }
        -1
    }
}